File: client-scalaz/src/main/scala/org/openapitools/client/api/Leg.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime

import java.time.OffsetDateTime

import Leg._

case class Leg (
  id: Option[Long],
active: Option[Boolean],
valid: Option[Boolean],
owner: Option[Account],
parentOwner: Option[Account],
application: Option[Application],
deviceId: Option[String],
distance: Option[Double],
duration: Option[Long],
startLatitude: Option[Double],
startLongitude: Option[Double],
startAltitude: Option[Double],
startDate: Option[OffsetDateTime],
endLatitude: Option[Double],
endLongitude: Option[Double],
endAltitude: Option[Double],
endDate: Option[OffsetDateTime],
`type`: Option[String],
retailerLocationId: Option[Long],
floorId: Option[String],
steps: Option[List[Step]])

object Leg {
  import DateTimeCodecs._

  implicit val LegCodecJson: CodecJson[Leg] = CodecJson.derive[Leg]
  implicit val LegDecoder: EntityDecoder[Leg] = jsonOf[Leg]
  implicit val LegEncoder: EntityEncoder[Leg] = jsonEncoderOf[Leg]
}

File: client-clojure/src/sirqul_io_t_platform/specs/leg.clj
(ns sirqul-io-t-platform.specs.leg
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [sirqul-io-t-platform.specs.account :refer :all]
            [sirqul-io-t-platform.specs.account :refer :all]
            [sirqul-io-t-platform.specs.application :refer :all]
            [sirqul-io-t-platform.specs.step :refer :all]
            )
  (:import (java.io File)))


(def leg-data
  {
   (ds/opt :id) int?
   (ds/opt :active) boolean?
   (ds/opt :valid) boolean?
   (ds/opt :owner) account-spec
   (ds/opt :parentOwner) account-spec
   (ds/opt :application) application-spec
   (ds/opt :deviceId) string?
   (ds/opt :distance) float?
   (ds/opt :duration) int?
   (ds/opt :startLatitude) float?
   (ds/opt :startLongitude) float?
   (ds/opt :startAltitude) float?
   (ds/opt :startDate) inst?
   (ds/opt :endLatitude) float?
   (ds/opt :endLongitude) float?
   (ds/opt :endAltitude) float?
   (ds/opt :endDate) inst?
   (ds/opt :type) string?
   (ds/opt :retailerLocationId) int?
   (ds/opt :floorId) string?
   (ds/opt :steps) (s/coll-of step-spec)
   })

(def leg-spec
  (ds/spec
    {:name ::leg
     :spec leg-data}))
